### PR TITLE
Update nix flake

### PR DESCRIPTION
**Pull request by Sicheng-Pan**
This PR updates the flake file and its lock file so that the latest commit can be built in a Nix environment.

### CODE REVIEW

File: man/nanoarrow.Rd
[the remark this answers is not in the document]
Comment (Sicheng-Pan) [reply man/nanoarrow.Rd:1]: Thanks for the notice!

**Comment by Sicheng-Pan** [comment]
@eitsupi I will take a look at #228

**Comment by eitsupi** [comment]
Ok, thanks.
I merge this then will merge #228.

**Review by eitsupi — APPROVED** [review]
The CI failure is not related to this change.

**Pull request merged**